Include/internal/pycore_regions.h

```diff
@@ -83,6 +89,32 @@ int _PyRegion_is_closed(PyObject* region);
 int _PyCown_release(PyObject *self);
 int _PyCown_is_released(PyObject *self);
 
+
+#ifdef _Py_TYPEOF
+#define Py_CLEAR_OBJECT_FIELD(op, field) \
+    do { \
+        _Py_TYPEOF(op)* _tmp_field_ptr = &(field); \
+        _Py_TYPEOF(op) _tmp_old_field = (*_tmp_field_ptr); \
+        if (_tmp_old_field != NULL) { \
+            *_tmp_field_ptr = _Py_NULL; \
+            Py_REGIONREMOVEREFERENCE(op, _tmp_old_field); \
+            Py_DECREF(_tmp_old_field); \
+        } \
+    } while (0)
+#else
+#define Py_CLEAR_OBJECT_FIELD(op, field) \
+    do { \
+        PyObject **_tmp_field_ptr = _Py_CAST(PyObject**, &(op)); \
+        PyObject *_tmp_old_field = (*_tmp_field_ptr); \
+        if (_tmp_old_field != NULL) { \
```

Comment:
Now they both do a NULL check, is this intentional?

Reply:
Yeah, I don't really understand why they have two versions, one is using memcpy to overwrite the pointer.   There is not a Py_XCLEAR, so I wasn't sure about adding one for us.